tests/lib/nested.sh

```diff
@@ -371,50 +391,53 @@ create_nested_core_vm(){
                 EXTRA_SNAPS="$EXTRA_SNAPS --snap $mysnap"
             done
 
-            if [ "$BUILD_SNAPD_FROM_CURRENT" = "true" ]; then
+            if [ "$NESTED_BUILD_SNAPD_FROM_CURRENT" = "true" ]; then
                 if is_core_16_nested_system; then
-                    repack_snapd_deb_into_core_snap "$WORK_DIR"
-                    EXTRA_FUNDAMENTAL="$EXTRA_FUNDAMENTAL --snap $WORK_DIR/core-from-snapd-deb.snap"
+                    repack_snapd_deb_into_core_snap "$NESTED_ASSETS_DIR"
+                    EXTRA_FUNDAMENTAL="$EXTRA_FUNDAMENTAL --snap $NESTED_ASSETS_DIR/core-from-snapd-deb.snap"
 
                 elif is_core_18_nested_system; then
-                    repack_snapd_deb_into_snapd_snap "$WORK_DIR"
-                    EXTRA_FUNDAMENTAL="$EXTRA_FUNDAMENTAL --snap $WORK_DIR/snapd-from-deb.snap"
+                    repack_snapd_deb_into_snapd_snap "$NESTED_ASSETS_DIR"
+                    EXTRA_FUNDAMENTAL="$EXTRA_FUNDAMENTAL --snap $NESTED_ASSETS_DIR/snapd-from-deb.snap"
 
                 elif is_core_20_nested_system; then
                     snap download --basename=pc-kernel --channel="20/edge" pc-kernel
-                    uc20_build_initramfs_kernel_snap "$PWD/pc-kernel.snap" "$WORK_DIR/image"
+                    uc20_build_initramfs_kernel_snap "$PWD/pc-kernel.snap" "$NESTED_ASSETS_DIR"
 
                     # Get the snakeoil key and cert
+                    local KEY_NAME SNAKEOIL_KEY SNAKEOIL_CERT
                     KEY_NAME=$(get_snakeoil_key)
                     SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
                     SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
 
                     # Prepare the pc kernel snap
-                    KERNEL_SNAP=$(ls "$IMAGE_DIR"/pc-kernel_*.snap)
-                    KERNEL_UNPACKED="$IMAGE_DIR"/kernel-unpacked
+                    local KERNEL_SNAP KERNEL_UNPACKED
+                    KERNEL_SNAP=$(ls "$NESTED_ASSETS_DIR"/pc-kernel_*.snap)
+                    KERNEL_UNPACKED="$NESTED_ASSETS_DIR"/kernel-unpacked
                     unsquashfs -d "$KERNEL_UNPACKED" "$KERNEL_SNAP"
                     sbattach --remove "$KERNEL_UNPACKED/kernel.efi"
                     sbsign --key "$SNAKEOIL_KEY" --cert "$SNAKEOIL_CERT" "$KERNEL_UNPACKED/kernel.efi"  --output "$KERNEL_UNPACKED/kernel.efi"
```

Comment:
this change isn't necessary here, but just as an FYI, `uc20_build_initramfs_kernel_snap` already re-signs the kernel.efi, so this isn't necessary, but https://github.com/snapcore/snapd/pull/9208 already fixes this

Reply:
Thanks, I already reviewed that one :)